A company started to invest in digital marketing as a new way of their product
promotions.For that they collected data and decided to carry out a study on it.

● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Help the company to carry out their study with the help of data provided.

In [1]:
#importing libraries

import numpy as np
import pandas as pd
import scipy.stats as stat


In [2]:
#reading and checking data

data=pd.read_csv('Sales_add.csv')
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [3]:
#checking for null values

data.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [4]:
#informations of data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 680.0+ bytes




# ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

Null Hypothesis, 

H0 :There is no increase in sales after stepping into digital marketing

Alternative Hypothesis,

H1 :There is an increase in sales after stepping into digital marketing

In [5]:
test1=stat.ttest_rel(data["Sales_before_digital_add(in $)"],data["Sales_After_digital_add(in $)"])
print(test1)

Ttest_relResult(statistic=-12.09070525287017, pvalue=6.336667004575778e-11)



Since the p value is lesser than 0.05 ,we can reject the null hypothesis,
There is an increase in sales after stepping into digital marketing

# ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Null Hypothesis 

H0:There is no dependency between the features "Region" and "Manager"

Alternative Hypothesis

H1:There is dependency between the features "Region" and "Manager"

In [6]:
crosstab = pd.crosstab(data["Region"],data["Manager"])
crosstab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [7]:
stat.chi2_contingency(crosstab)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

p_value=0.5493991051158094

Since p_value is greater than 0.05 (alpha value) ,we cannot reject the null hypothesis , There is no dependency between the features "Region" and "Manager" 